app/services/ccms/submitters/obtain_document_id_service.rb

```diff
@@ -37,26 +42,33 @@ def populate_documents
         end
       end
 
-      def request_document_ids
-        submission.submission_documents.each do |document|
-          tx_id = document_id_requestor.transaction_request_id
-          @response = document_id_requestor.call
-          document.ccms_document_id = Parsers::DocumentIdResponseParser.new(tx_id, @response).document_id
-          document.status = :id_obtained
-          document.save!
-          create_history('applicant_ref_obtained', 'document_ids_obtained', xml_request, @response) if submission.save!
-        rescue CcmsError => e
-          document.status = :failed
-          raise CcmsError, e
-        end
+      def request_document_id(document)
+        document_id_requestor = document_id_requestor(document.document_type)
+        response = update_document_and_return_response(document, document_id_requestor)
+        document.save!
+        create_history('applicant_ref_obtained', 'document_ids_obtained', document_id_requestor.formatted_xml, response) if submission.save!
+      rescue CcmsError => e
+        document.status = :failed
+        document.save!
+        create_ccms_failure_history('applicant_ref_obtained', e, document_id_requestor.formatted_xml)
```

Comment:
Could you reuse handle_exception method here (it's set in the parent class BaseSubmissionService)